sktime/forecasting/model_evaluation/_functions.py

```diff
@@ -597,13 +607,23 @@ def gen_y_X_train_test(y, X, cv, cv_X):
         # Run temporal cross-validation sequentially
         results = []
         for x in enumerate(yx_splits):
-            is_first = x[0] == 0  # first iteration
+            i = x[0]
+            is_first = i == 0  # first iteration
+            if is_first:
+                callbacks.on_iteration_start()
             if strategy == "update" or (strategy == "no-update_params" and is_first):
-                result, forecaster = _evaluate_window(x, _evaluate_window_kwargs)
+                result, y_pred, forecaster = _evaluate_window(
+                    x, _evaluate_window_kwargs
+                )
                 _evaluate_window_kwargs["forecaster"] = forecaster
```

Comment:
@fkiraly any idea why are we returning the forecaster and adding it to `_evaluate_window_kwargs `? It seems this dict is not used anymore down the function (because the if/else excludes the execution of the rest of the code).

It is a bit puzzling to me to have a function `_evaluate_window` which returns a different number of objects depending on the `strategy` parameter.